pkg/service/backup/purger.go

```diff
@@ -23,31 +24,39 @@ import (
 // staleTags returns collection of snapshot tags for manifests that are "stale".
 // That is:
 // - temporary manifests,
+// - manifests over task days retention days policy,
 // - manifests over task retention policy,
 // - manifests older than threshold if retention policy is unknown.
-func staleTags(manifests []*ManifestInfo, policy map[uuid.UUID]int, threshold time.Time) *strset.Set {
+func staleTags(manifests []*ManifestInfo, policy map[uuid.UUID]Retention, threshold time.Time) *strset.Set {
```

Comment:
I think we replace `policy map[uuid.UUID]Retention, threshold time.Time` with a `func(uuid.UUID) Retention`.

Reply:
Added `type RetentionFunc func(uuid.UUID) Retention` which is now a property of the backup target in place of the retention map.

The function checks for policies in the retention map and if none exists returns a default of 30 retention days, which was the old threshold to delete backups with no associated task.